p2p/libp2p/identityProvider.go

```diff
@@ -0,0 +1,184 @@
+package libp2p
+
+import (
+	"context"
+	"encoding/hex"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/p2p/data"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/multiformats/go-multiaddr"
+)
+
+const authProtocolID = "elrond-node-auth/0.0.1"
+const maxBytesToReceive = 1024
+
+type identityProvider struct {
+	host                     host.Host
+	networkShardingCollector p2p.NetworkShardingCollector
+	signerVerifier           p2p.SignerVerifier
+	marshalizer              p2p.Marshalizer
+	receiveTimeout           time.Duration
+}
+
+// NewIdentityProvider creates a wrapper over libp2p's network.Notifiee implementation
+// that is able to do a supplemental authentication against each peer connected
+func NewIdentityProvider(
+	h host.Host,
+	networkShardingCollector p2p.NetworkShardingCollector,
+	signerVerifier p2p.SignerVerifier,
+	marshalizer p2p.Marshalizer,
+	receiveTimeout time.Duration,
+) (*identityProvider, error) {
+
+	if h == nil {
```

Comment:
check ifNil ?

Reply:
host.Host does not have the function IsInterfaceNil (host.Host is defined in libp2p's libraries)